src/Events.php

```diff
@@ -156,6 +157,13 @@ protected function possibly_add_notes() {
 		UpdateStoreDetails::possibly_add_note();
 	}
 
+	/**
+	 * Deletes notes that should be deleted.
+	 */
+	protected function possibly_delete_notes() {
```

Comment:
💯

Reply:
> Having a single function might save some headaches down the road if we plan to add/remove a number of notes. Naming needs work, but something along the lines of `should_have_note()`.

I had considered the idea of something like a `should_note_exist()` method, yeah.

The biggest problem I have with making this feel right (whatever that means) is that the "possibly" in the existing `possibly_delete_note()` methods is just about "Hey maybe there's no note to delete" rather than "Hey maybe the note shouldn't be deleted at this time."

> Setting the `is_deleted` flag is a better idea if we want the note to not show up again, but I would think this note should (re)appear if the feature is available in the currently active version.

I almost always prefer a soft delete to a hard one, but I agree the note should reappear when it becomes relevant again. We could add some logic to `possibly_add_note()` that reverses a soft delete, but since soft delete is what happens when the user manually dismisses the note (I haven't checked that on NavigationNudge yet, but that's what happened on the refund and return note), we definitely don't want to create a zombie note situation.

Come to think of it, we kind of have a zombie note situation now, though it's kind of a long walk to get there. Should I add  logic to `delete_if_not_supported()` to keep soft-deleted notes from being hard-deleted, so they don't get recreated later?

Reply:
>Come to think of it, we kind of have a zombie note situation now, though it's kind of a long walk to get there. Should I add logic to delete_if_not_supported() to keep soft-deleted notes from being hard-deleted, so they don't get recreated later?

That's a great point.  I'll leave a comment below.